### DISCUSSION TITLE
insecureHTTPParser in fetch RequestInit options

### DISCUSSION THREAD

**Discussion by witia87** [nodejs/help#4746] [Ideas]
Hey!
Since the two things:
1. native `fetch`, as well as
2. `insecureHTTPParser` option available on `http/https.requests`

have been introduced in latest versions of node.js, it would be fitting so see this flag available also on `fetch` options.
Our teem would gladly see this improvement implemented and also you can see how this feature was requested/implemented in node-fetch v3:
https://github.com/node-fetch/node-fetch/issues/724 
https://github.com/node-fetch/node-fetch/blob/7b86e946b02dfdd28f4f8fca3d73a022cbb5ca1e/src/request.js#L139

Do you think it's possible to get this feature on node.js native fetch?

**Comment by krzysztof-o** [comment]
👍 I've had the same problem. Using [--insecure-http-parser](https://nodejs.org/docs/latest-v12.x/api/cli.html#cli_insecure_http_parser) is annoying as it prints error on stdout.

**Comment by bnoordhuis** [comment]
fetch() is a web standard, insecureHTTPParser is not. You can plug your own agent into fetch() though.

**Reply by witia87** [reply]
Hey, could you tell a bit more on how could I plug this agent to help with insecure HTTP headers?

**Reply by bnoordhuis** [reply]
`npm add undici`, then:
```js
import {Agent} from 'undici'
const dispatcher = new Agent({/*...*/})
fetch(url, {dispatcher})
```

**Reply by witia87** [reply]
thank you, I'll give it a try!
